src/main/java/sg/ncl/MainController.java

```diff
@@ -1731,16 +1734,14 @@ public String checkJoinTeamInfo(
                 MyErrorResource error = objectMapper.readValue(responseBody, MyErrorResource.class);
                 ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
 
-                if (exceptionState == ExceptionState.TEAM_NOT_FOUND_EXCEPTION) {
-                    log.warn("join team request : team name error");
-                    redirectAttributes.addFlashAttribute(MESSAGE, "Team name does not exists.");
-                }
-                else if(exceptionState == ExceptionState.DETERLAB_OPERATION_FAILED_EXCEPTION) {
-                    log.warn("join team request: operation failed on DeterLab");
-                    redirectAttributes.addFlashAttribute(MESSAGE, "Error: " + (error.getMessage().contains("unknown error")? ERR_SERVER_OVERLOAD : error.getMessage()));
-                }
-                else {
-                    log.warn("join team request : some other failure");
+                if (checkUserException(exceptionState, error) != null || checkDeterlabException(exceptionState, error) != null) {
+                    log.info("Join team request : " + error.getMessage());
+                    redirectAttributes.addFlashAttribute(MESSAGE, error.getMessage());
+                } else if (exceptionState == TEAM_NOT_FOUND_EXCEPTION) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This branch's code block is the same as the block for the branch on line 1737. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1871)